src/odemis/gui/comp/text.py

```diff
@@ -618,6 +618,78 @@ def _step_from_range(min_val, max_val):
         logging.exception(msg)
 
 
+class PatternValidator(ValidatorClass):
+
+    def __init__(self, pattern):
+        """ pattern (str): regex pattern of allowed entries """
+        super().__init__()
+        self.Bind(wx.EVT_CHAR, self.on_char)
+        self.entry_pattern = pattern
+
+    def _is_valid_value(self, val):
+        """ Validate the given value
+
+        Args:
+            val (str):
+
+        Returns:
+            (boolean): True if the given string is valid
+
+        """
+
+        entry_pattern = re.compile(self.entry_pattern)
+        return bool(re.match(entry_pattern, val)) and entry_pattern.match(val).end() == len(val)
```

Comment:
You can use `pattern.fullmatch()` to check that. (Since Python 3.4, which is fine for us as we have Python 3.5+)